### ISSUE TITLE
Tpetra initialize with "--help"

### ISSUE THREAD

**Issue by kyungjoo-kim** [trilinos/Trilinos#2303]
An example code uses tpetra init/finalize as follows: 
```
Tpetra::initialize(&argc, &argv);
// some Teuchos command line input also used
// printing Teuchos command line help is guarded with mpi comm. 
Tpetra::finalize();
```
When we run this code with "--help", this prints multiple kokkos command line help and warning related to openmp bindings. 

This is not a critical problem but bothers users. 

@trilinos/tpetra 

## Expectations

Kokkos warnings and help should be printed by the root mpi rank only.

## Current Behavior

Kokkos warnings and help are printed by all ranks.

**Comment by mhoemmen** [comment]
@kyungjoo-kim There are two ways to fix this:
  1. Add an overload of `Kokkos::initialize` that takes an `std::ostream&` to which to write help information.
  2. Override `std::cout` temporarily inside `Tpetra::initialize`.

You can do (2) safely with RAII, by using `rdbuf()`.  See the following example (`buffer_restore`):

https://stackoverflow.com/questions/18975512/how-to-override-cout-in-c

This is not hard to do.  Would you like to submit a pull request that does it?

**Comment by mhoemmen** [comment]
I pushed a fix to develop.

**Comment by ibaned** [comment]
Thanks @mhoemmen !
@dsunder was working on a proper fix in Kokkos for this, but its hasn't made it in yet.

**Comment by ibaned** [comment]
We'll let you know when its in so you don't have to do this workaround in Tpetra

**Comment by mhoemmen** [comment]
@ibaned I already did it in Tpetra :-D .  It's not really Kokkos' job to know about MPI processes or communicators, at least not currently.  Imagine, for example, an MPMD application that only starts Kokkos on a subset of processes.  I wouldn't want to assume that Process 0 on `MPI_COMM_WORLD` should print, because that process may not be in the communicator that starts Kokkos.

**Comment by ibaned** [comment]
@mhoemmen I personally think that for reasons like that, Kokkos should have a proper MPI dependency. It also needs to decide which GPU on the node to select, which it currently does by getting MPI rank information from environment variables as opposed to an API

**Comment by mhoemmen** [comment]
@ibaned I don't have very strong feelings about this, but it's not so hard for Tpetra::initialize to do the right thing (it already does, in fact).

I spent some free / airplane time last week on #2372.  My updates will include a test for exactly the MPMD case of `Tpetra::initialize` and `Tpetra::finalize` only being called on a subcommunicator of `MPI_COMM_WORLD` that does not include Process 0 in `MPI_COMM_WORLD`.

**Comment by mhoemmen** [comment]
FYI my #2372 fix PR is #2413.
